test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -533,7 +533,7 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                         "GET / HTTP/1.1",
                         "",
                         "");
-                    await connection.ReceiveEnd(
+                    await connection.ReceiveForcedEnd(
```

Comment:
Wouldn't `ReceiveEnd` send the FIN after it received all the expected data now? I expect `ReceiveForcedEnd` would still take a while since it's waiting for the connection to close with an IOException or otherwise. Unlike `ReceiveEnd`, `ReceiveForcedEnd` hasn't been modified to send a FIN.